src/main/java/algorithmic_task/service/Impl/SimilarityServiceImpl.java

```diff
@@ -35,21 +36,14 @@ public boolean areSimilar(String sentence1, String sentence2) {
 
         if (s1.size() != s2.size() || s1.size() == 0) return false;
 
-        Map<String, Integer> mapDictionary = new HashMap<>();
-
-        for (int i = 0; i < ((List<Set<String>>) dictionary.getDictionary()).size(); i++) {
-            for (String dic : ((List<Set<String>>) dictionary.getDictionary()).get(i)) {
-                mapDictionary.put(dic, i);
-            }
-        }
+        Map<String, Integer> dictionary = dictionaryService.getDictionary();
 
         for (int i = 0; i < s1.size(); i++)
-            if (!mapDictionary.getOrDefault(s1.get(i), -1).equals(mapDictionary.getOrDefault(s2.get(i), -2)))
+            if (!dictionary.getOrDefault(s1.get(i), -1).equals(dictionary.getOrDefault(s2.get(i), -2)))
```

Comment:
what is -1 and -2?

Reply:
If the word from the sentence isn't in dictionary method should return Integer, but if both sentences aren't in dictionary this Integers should be different. That is why I use -1 and -2. Probably it's not the best solution but I don't know how to do this better and I didn't want to use null. For sure I should describe that in documentation.